### ISSUE TITLE
[node compat] Unable to import cjs exports set with `defineProperty`

### ISSUE THREAD

**Issue by thislooksfun** [oven-sh/bun#4432]
### What version of Bun is running?

0.8.1+16b4bf341acc0f4804f0b6bdf5298c180cd00366

### What platform is your computer?

Darwin 22.5.0 x86_64 i386

### What steps can reproduce the bug?

Create the following two files:

```js
/// File: source.cjs
Object.defineProperty(exports, 'hello', {
  value: 'world',
});

/// File: sink.mjs
import { hello } from './source.cjs';
console.log(hello);
```

And run `bun sink.mjs`

### What is the expected behavior?

It should print `world`.

### What do you see instead?

```
SyntaxError: Import named 'hello' not found in module '<path/to/sink.mjs>'.
```

### Additional information

- Running `node sink.mjs` works as expected.
- Replacing the `defineProperty` with just `export.hello = "world"` works as expected.
- Real-world example: [triple-beam](https://www.npmjs.com/package/triple-beam).

**Comment by Jarred-Sumner** [comment]
This bug happens because `enumerable` is not set to true and Bun currently ignores non-enumerable properties when converting CJS to ESM 

We should probably change that to include non-enumerable properties that have no side effects

**Comment by thislooksfun** [comment]
Updated the description with an example of this pattern being used in the wild.

**Comment by nomoon** [comment]
[Winston](https://github.com/winstonjs/winston)'s sharable symbol constants also does this. https://github.com/winstonjs/triple-beam/blob/master/index.js

**Comment by Kleywalker** [comment]
I use winston and got the error message `SyntaxError: Import named 'LEVEL' not found in module './node_modules/triple-beam/index.js'.
      at processTicksAndRejections (:55:76)`. Are there any news when this bug will be fixed? 😊 

I use revision 1.0.4+745b6b94ee56cad24d475799690cc9a89957d15b

**Comment by imcm7** [comment]
Import named 'Routes' not found in module '/home/Project/node/nuxt/interfaces/router/index.ts'

index file

```
export * from '~/server/libraries/core/interfaces/router/routes.interface';
```

**Comment by Sfinx** [comment]
bun 1.0.14, pkg https://github.com/Oaxoa/Keysme, enumerable is set:

keysme/lib/index.js
```
Object.defineProperty(exports, "__esModule", {
  value: true
});

Object.defineProperty(exports, "getKey", {
  enumerable: true,
  get: function get() {
    return _utils.getKey;
  }
});
```

keysme/index.js
```
import {getKey, checkHotkey, registerHotkey, deregisterHotkey, getRegisteredHotkeys} from './lib';
export {getKey, checkHotkey, registerHotkey, deregisterHotkey, getRegisteredHotkeys};
```

bun error:

```
'getKey' is not exported by keysme/lib/index.js, imported by keysme/index.js
```

**Comment by Jarred-Sumner** [comment]
This will be fixed in Bun v1.1.27

sorry it took so long
